client/my-sites/plans/jetpack-plans/product-store/utils/isolate-popular-items.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Converts an array of items (objects) into tuple
+ * with first item of the tuple being the array of popular items
+ * and second item being an array of the rest of the items.
+ */
+export const isolatePopularItems = < Item extends { productSlug: string } >(
+	allItems: Array< Item >,
+	popularItemSlugs: Array< Item[ 'productSlug' ] >
+) => {
+	const popularItems: typeof allItems = [];
+	const otherItems: typeof allItems = [];
+
+	for ( const item of allItems ) {
+		if ( popularItemSlugs.includes( item.productSlug ) ) {
+			popularItems.push( item );
+		} else {
+			otherItems.push( item );
+		}
+	}
+
+	return [ popularItems, otherItems ] as const;
```

Comment:
Just out of curiousity why have you decided to use `tuple` here instead of an object? I would like to know more for myself on when best to use it.

Reply:
Not a specific reason but I saw that the same pattern was used in `client/my-sites/plans/jetpack-plans/product-grid/index.tsx`, so I did the same.